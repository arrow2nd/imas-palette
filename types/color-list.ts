export type ColotListType = {
  name: string
  hex: string
}
