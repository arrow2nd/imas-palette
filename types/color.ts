export type ColorType = {
  rgb: string
  hsv: string
  hex: string
  similar: string
}
