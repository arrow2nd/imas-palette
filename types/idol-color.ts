export type IdolColorType = {
  rgb: string
  hsv: string
  hex: string
  similar: string
}
