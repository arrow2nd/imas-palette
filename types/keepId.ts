export type KeepId = {
  list: string[]
  add: (addId: string) => void
  remove: (removeId: string) => void
}
