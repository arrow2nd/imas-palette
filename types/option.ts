export type OptionType = {
  title: string
  value: string
}
