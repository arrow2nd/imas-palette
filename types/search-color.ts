export type SearchColor = {
  name: string
  hex: string
}
