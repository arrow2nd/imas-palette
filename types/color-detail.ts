import { Color } from './color'

export type ColorDetail = {
  id: string
  name: string
  nameSuppl: string
  nameKana: string
  brand: string
  color: Color
}
