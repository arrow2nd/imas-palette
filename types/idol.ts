import { color } from './color'

export type idol = {
  nameJa: string
  nameEn: string
  color: color
}
