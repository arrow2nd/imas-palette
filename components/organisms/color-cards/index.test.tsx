import { render } from '@testing-library/react'

import { ColorDetail } from 'types/color-detail'

import ColorCards from './index'

describe('ColorCards', () => {
  const props = {
    className: '',
    keepIdList: [],
    onAddKeepId: jest.fn(),
    onRemoveKeepId: jest.fn()
  }

  test('カードが無いときの表示が正しいか', () => {
    const { getByText } = render(<ColorCards {...props} items={[]} />)
    expect(getByText('みつかりませんでした...')).toBeTruthy()
  })

  test('見た目が変化していないか', () => {
    const colorDetails: ColorDetail[] = [
      {
        id: 'yuika_mitsumine_shinycolors',
        name: '三峰結華',
        nameSuppl: 'Yuika Mitsumine',
        nameKana: 'みつみねゆいか',
        brand: 'ShinyColors',
        color: {
          rgb: 'rgb(59, 145, 196)',
          hsv: 'hsv(202, 70, 77)',
          hex: '#3B91C4',
          similar: '#00ffff'
        }
      },
      {
        id: 'iori_minase_765as',
        name: '水瀬伊織',
        nameSuppl: 'Iori Minase',
        nameKana: 'みなせいおり',
        brand: '765AS',
        color: {
          rgb: 'rgb(253, 153, 225)',
          hsv: 'hsv(317, 40, 99)',
          hex: '#FD99E1',
          similar: '#ffffff'
        }
      },
      {
        id: 'miyu_mifune_cinderellagirls',
        name: '三船美優',
        nameSuppl: 'Miyu Mifune',
        nameKana: 'みふねみゆ',
        brand: 'CinderellaGirls',
        color: {
          rgb: 'rgb(1, 170, 165)',
          hsv: 'hsv(178, 99, 67)',
          hex: '#01AAA5',
          similar: '#00ffff'
        }
      }
    ]

    const { container } = render(<ColorCards {...props} items={colorDetails} />)

    expect(container).toMatchSnapshot()
  })
})
