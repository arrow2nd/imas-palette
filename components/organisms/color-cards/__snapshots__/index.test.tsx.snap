// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCards 見た目が変化していないか 1`] = `
<div>
  <div
    class="flex flex-row flex-wrap justify-center "
  >
    <div
      class="w-96 m-2 p-6 rounded-lg shadow-md bg-white h-64"
      data-testid="color-card"
    >
      <div
        class="flex flex-row items-center"
      >
        <div
          class="flex-none w-16 h-16 rounded-xl border border-gray-200"
          style="background-color: rgb(59, 145, 196);"
        />
        <div
          class="flex-1 ml-6"
          data-testid="name-label"
        >
          <p
            class="text-lg leading-6"
          >
            三峰結華
          </p>
          <p
            class="text-sm tracking-wide"
          >
            Yuika Mitsumine
          </p>
        </div>
        <button
          class="px-3 py-1 text-center items-center border border-natural-black text-white hover:text-natural-black bg-natural-black hover:bg-natural-white rounded-lg transition-colors inline-flex text-sm ml-4"
          data-testid="keep-button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
            />
          </svg>
          <span
            class="ml-2 tracking-wide"
          >
            Keep
          </span>
        </button>
      </div>
      <div
        class="mt-6"
      >
        <div
          class="flex flex-row items-center "
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            RGB
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            rgb(59, 145, 196)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HSV
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            hsv(202, 70, 77)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HEX
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            #3B91C4
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="w-96 m-2 p-6 rounded-lg shadow-md bg-white h-64"
      data-testid="color-card"
    >
      <div
        class="flex flex-row items-center"
      >
        <div
          class="flex-none w-16 h-16 rounded-xl border border-gray-200"
          style="background-color: rgb(253, 153, 225);"
        />
        <div
          class="flex-1 ml-6"
          data-testid="name-label"
        >
          <p
            class="text-lg leading-6"
          >
            水瀬伊織
          </p>
          <p
            class="text-sm tracking-wide"
          >
            Iori Minase
          </p>
        </div>
        <button
          class="px-3 py-1 text-center items-center border border-natural-black text-white hover:text-natural-black bg-natural-black hover:bg-natural-white rounded-lg transition-colors inline-flex text-sm ml-4"
          data-testid="keep-button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
            />
          </svg>
          <span
            class="ml-2 tracking-wide"
          >
            Keep
          </span>
        </button>
      </div>
      <div
        class="mt-6"
      >
        <div
          class="flex flex-row items-center "
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            RGB
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            rgb(253, 153, 225)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HSV
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            hsv(317, 40, 99)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HEX
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            #FD99E1
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="w-96 m-2 p-6 rounded-lg shadow-md bg-white h-64"
      data-testid="color-card"
    >
      <div
        class="flex flex-row items-center"
      >
        <div
          class="flex-none w-16 h-16 rounded-xl border border-gray-200"
          style="background-color: rgb(1, 170, 165);"
        />
        <div
          class="flex-1 ml-6"
          data-testid="name-label"
        >
          <p
            class="text-lg leading-6"
          >
            三船美優
          </p>
          <p
            class="text-sm tracking-wide"
          >
            Miyu Mifune
          </p>
        </div>
        <button
          class="px-3 py-1 text-center items-center border border-natural-black text-white hover:text-natural-black bg-natural-black hover:bg-natural-white rounded-lg transition-colors inline-flex text-sm ml-4"
          data-testid="keep-button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
            />
          </svg>
          <span
            class="ml-2 tracking-wide"
          >
            Keep
          </span>
        </button>
      </div>
      <div
        class="mt-6"
      >
        <div
          class="flex flex-row items-center "
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            RGB
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            rgb(1, 170, 165)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HSV
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            hsv(178, 99, 67)
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="flex flex-row items-center mt-4"
        >
          <div
            class="flex-none w-16 py-1 text-center text-sm tracking-wide rounded-lg bg-gray-200"
          >
            HEX
          </div>
          <button
            class="flex-1 flex flex-row justify-end items-center text-sm tracking-wide cursor-pointer"
            data-testid="copy-button"
            title="クリックでコピー"
          >
            #01AAA5
            <span
              class="ml-2"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
